userspace/Makefile

```diff
@@ -23,20 +24,23 @@ KERNEL_OBJ = kernel.o
 .PHONY: all
 all: $(INITRD)
 
-$(INITRD): $(USERSPACE_BINS)
+$(INITRD): $(wildcard $(DUMBFS_DATA)/*)
```

Comment:
I think it's annoying to have an other directory in the kernel for such a little impact (debatable). In addition, the directory is named like the filesystem, and that should be transparent to the user.

It would be cleaner to use the `$(USERSPACE_BINS)` variable to know which files should be placed inside the initial ramdisk, and add an option to manually add plain-text files as a dependency of `$(INITRD)`.

Thus, i suggest to:
  - Remove any references to `dumbfs_files` and $(DUMBFS_DATA)
  - Make `../scripts/initrd.sh` take the files that should be added in the ramdisk as parameters
  - Give him `$<` as parameters.

What do you think ?

Reply:
This will also get rid of the symbolic link. I think it is indeed the cleanest solution.